src/tests/nativeaot/SmokeTests/Reflection/Reflection.cs

```diff
@@ -2594,10 +2594,14 @@ private static async void ReflectionInLambdaAsync()
 
         public static void Run()
         {
+            // Thread.Start not supported for Wasm.
             ReflectionInLambda();
             ReflectionInLocalFunction();
+            // Thread.Start not supported for Wasm.
+#if !CODEGEN_WASM
             ReflectionInAsync();
             ReflectionInLambdaAsync();
```

Comment:
Does this not work on `Browser`? It should, theoretically at least.

Reply:
This is the problem
```
        To run the test:
        > set CORE_ROOT=C:\github\runtimelab\artifacts\tests\coreclr\Browser.wasm.Debug\Tests\Core_Root
        > C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection_FromUsage\Reflection_FromUsage.cmd
        Stack Trace:
          C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\TestWrappers\nativeaot.SmokeTests\nativeaot.SmokeTests.XUnitWrapper.cs(1139,0): a
  t nativeaot_SmokeTests._Reflection_Reflection_FromUsage_Reflection_FromUsage_._Reflection_Reflection_FromUsage_Reflection_FromUsage_cmd()
             at System.RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
             at System.Reflection.MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
        Output:
          Unhandled exception. System.TypeInitializationException: A type initializer threw an exception. To determine which type, inspect the InnerExcepti
  on's StackTrace property.
           ---> System.PlatformNotSupportedException: Operation is not supported on this platform.
              at Reflection.FromUsage.wasm.System.Runtime.EH.DispatchException (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[11481]:0x8b6ca
  e)
              at Reflection.FromUsage.wasm.System.Runtime.EH.RhpThrowEx (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[14615]:0xb2921d)
              at Reflection.FromUsage.wasm.System.Threading.TimerQueue..ctor (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[7479]:0x5a9bd7)
              at Reflection.FromUsage.wasm.System.Threading.TimerQueue.CreateTimerQueues (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[2344
  3]:0x11ff74e)
              at Reflection.FromUsage.wasm.System.Threading.TimerQueue..cctor (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[7098]:0x541419)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.ClassConstructorRunner.EnsureClassConstructorRun (wasm://wasm/Reflection.FromUsa
  ge.wasm-0a59e0ba:wasm-function[17878]:0xd7a24f)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.ClassConstructorRunner.CheckStaticClassConstructionReturnNonGCStaticBase (wasm:/
  /wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[8181]:0x6375e6)
              at Reflection.FromUsage.wasm..GetGCStaticBase.System.Threading.TimerQueue (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[31793
  ]:0x15a6e2f)
              at Reflection.FromUsage.wasm.System.Threading.TimerQueue.get.Instances (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[23036]:0
  x11bcaab)
              at Reflection.FromUsage.wasm.System.Threading.TimerQueueTimer..ctor.0 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[6553]:0x4
  d2a3a)
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.DelayPromise..ctor (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[225
  14]:0x1143f66)
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.Delay.5 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[6013]:0x455664
  )
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.Delay.4 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[21822]:0x10b1b
  ba)
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.Delay.3 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[2428]:0x1725a2
  )
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.TestCompilerGeneratedCode.ReflectionInAsync.d.6.MoveNext (wasm://wasm/Reflec
  tion.FromUsage.wasm-0a59e0ba:wasm-function[24579]:0x1315c81)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start<System..Canon> (wasm://wasm/Reflection.FromUsage.wa
  sm-0a59e0ba:wasm-function[2055]:0x12cdbb)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start<System..Canon> (wasm://wasm/Reflection.FromUsage.wa
  sm-0a59e0ba:wasm-function[24371]:0x12d93e2)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.TestCompilerGeneratedCode.ReflectionInAsync (wasm://wasm/Reflection.FromUsag
  e.wasm-0a59e0ba:wasm-function[1865]:0x10d089)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.TestCompilerGeneratedCode.Run (wasm://wasm/Reflection.FromUsage.wasm-0a59e0b
  a:wasm-function[24236]:0x12c6eff)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.Main (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[4987]:0x3
  90f03)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.Module..MainMethodWrapper (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[499
  1]:0x39124e)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.Module..StartupCodeMain (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[24205
  ]:0x12b8a9a)
              at Reflection.FromUsage.wasm.main (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[41]:0x1f60b)
              at C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Reflection.Fro
  mUsage.js:696:14
              at callMain (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Refl
  ection.FromUsage.js:7747:15)
              at doRun (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Reflect
  ion.FromUsage.js:7797:23)
              at run (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Reflectio
  n.FromUsage.js:7812:5)
              at runCaller (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Ref
  lection.FromUsage.js:7724:19)
              at removeRunDependency (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\
  native\Reflection.FromUsage.js:632:7)
              at receiveInstance (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\nati
  ve\Reflection.FromUsage.js:1031:5)
              at receiveInstantiationResult (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.Fro
  mUsage\native\Reflection.FromUsage.js:1049:5)

             --- End of inner exception stack trace ---
              at Reflection.FromUsage.wasm.System.Runtime.EH.DispatchException (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[11481]:0x8b6ca
  e)
              at Reflection.FromUsage.wasm.System.Runtime.EH.RhpThrowEx (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[14615]:0xb2921d)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.ClassConstructorRunner.EnsureClassConstructorRun (wasm://wasm/Reflection.FromUsa
  ge.wasm-0a59e0ba:wasm-function[17878]:0xd7a5e3)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.ClassConstructorRunner.CheckStaticClassConstructionReturnNonGCStaticBase (wasm:/
  /wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[8181]:0x6375e6)
              at Reflection.FromUsage.wasm..GetGCStaticBase.System.Threading.TimerQueue (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[31793
  ]:0x15a6e2f)
              at Reflection.FromUsage.wasm.System.Threading.TimerQueue.get.Instances (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[23036]:0
  x11bcaab)
              at Reflection.FromUsage.wasm.System.Threading.TimerQueueTimer..ctor.0 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[6553]:0x4
  d2a3a)
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.DelayPromise..ctor (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[225
  14]:0x1143f66)
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.Delay.5 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[6013]:0x455664
  )
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.Delay.4 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[21822]:0x10b1b
  ba)
              at Reflection.FromUsage.wasm.System.Threading.Tasks.Task.Delay.3 (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[2428]:0x1725a2
  )
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.TestCompilerGeneratedCode.ReflectionInAsync.d.6.MoveNext (wasm://wasm/Reflec
  tion.FromUsage.wasm-0a59e0ba:wasm-function[24579]:0x1315c81)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start<System..Canon> (wasm://wasm/Reflection.FromUsage.wa
  sm-0a59e0ba:wasm-function[2055]:0x12cdbb)
              at Reflection.FromUsage.wasm.System.Runtime.CompilerServices.AsyncVoidMethodBuilder.Start<System..Canon> (wasm://wasm/Reflection.FromUsage.wa
  sm-0a59e0ba:wasm-function[24371]:0x12d93e2)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.TestCompilerGeneratedCode.ReflectionInAsync (wasm://wasm/Reflection.FromUsag
  e.wasm-0a59e0ba:wasm-function[1865]:0x10d089)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.TestCompilerGeneratedCode.Run (wasm://wasm/Reflection.FromUsage.wasm-0a59e0b
  a:wasm-function[24236]:0x12c6eff)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.ReflectionTest.Main (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[4987]:0x3
  90f03)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.Module..MainMethodWrapper (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[499
  1]:0x39124e)
              at Reflection.FromUsage.wasm.Reflection.FromUsage.Module..StartupCodeMain (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[24205
  ]:0x12b8a9a)
              at Reflection.FromUsage.wasm.main (wasm://wasm/Reflection.FromUsage.wasm-0a59e0ba:wasm-function[41]:0x1f60b)
              at C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Reflection.Fro
  mUsage.js:696:14
              at callMain (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Refl
  ection.FromUsage.js:7747:15)
              at doRun (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Reflect
  ion.FromUsage.js:7797:23)
              at run (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Reflectio
  n.FromUsage.js:7812:5)
              at runCaller (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\native\Ref
  lection.FromUsage.js:7724:19)
              at removeRunDependency (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\
  native\Reflection.FromUsage.js:632:7)
              at receiveInstance (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.FromUsage\nati
  ve\Reflection.FromUsage.js:1031:5)
              at receiveInstantiationResult (C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection.Fro
  mUsage\native\Reflection.FromUsage.js:1049:5)

          Aborted(native code called abort())
          C:\github\runtimelab\artifacts\tests\coreclr\browser.wasm.Debug\nativeaot\SmokeTests\Reflection\Reflection_FromUsage\native\Reflection_FromUsage.
  js:128
                throw ex;
                ^
```

Reply:
I see. Well, let it be as it is.